magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleShownWithContext.kt

```diff
@@ -0,0 +1,41 @@
+package com.wealthfront.magellan.lifecycle
+
+import android.content.Context
+
+class LifecycleShownWithContext<V>(
+  val supplier: (Context) -> V
+) : LifecycleAware {
+
+  var data: V? = null
+    protected set
+
+  override fun show(context: Context) {
+    data = supplier(context)
+  }
+
+  override fun hide(context: Context) {
+    data = null
+  }
+}
+
+class LifecycleCreatedWithContext<V>(
+  val supplier: (Context) -> V
+) : LifecycleAware {
+
+  var data: V? = null
+    protected set
+
+  override fun create(context: Context) {
+    data = supplier(context)
+  }
+
+  override fun hide(context: Context) {
+    data = null
+  }
```

Comment:
What happens when it's `SHOWN` for a second time? This isn't symmetrical and will cause the `data` to be null after the first `hide` call.

For simplicity, should we just attach/detach the activity and views at the same time? That is, have a single `lifecycleWithContext` that attaches/detaches when `SHOWN`. We don't have a stable `Context` for the whole time we're `CREATED` anyway, so the current implementation is a bit tricky at best to use correctly.

Reply:
done. Consolidate to just use show/hide.